src/activities/Elsa.Activities.Rpa.Web/Extensions/SeleniumExtensions.cs

```diff
@@ -0,0 +1,355 @@
+                ? "+="
+                : direction == Direction.Left || direction == Direction.Up
+                    ? "-="
+                    : throw new Exception("Direction not supported");
+
+            var scrolling = direction == Direction.Right || direction == Direction.Left ? "scrollLeft" : "scrollTop";
+            remote.WrappedDriver.ExecuteJavaScript($"arguments[0].{scrolling} {scroll} {offset}", element);
+        }
+
+        public static void ScrollWindow(this IWebDriver driver, Direction direction, int offset)
+        {
+            var horizontalScroll = direction == Direction.Right
+              ? offset
+              : direction == Direction.Left
+                ? offset * -1
+                : 0;
+            var verticalScroll = direction == Direction.Down
+              ? offset
+              : direction == Direction.Up
+                ? offset * -1
+                : 0;
+            driver.ExecuteJavaScript($"window.scrollBy({horizontalScroll}, {verticalScroll});");
+        }
+
+        public static T GetProperty<T>(this IWebElement element, string name)
+        {
+            var remote = (RemoteWebElement)element;
+            return remote.WrappedDriver.ExecuteJavaScript<T>($"return arguments[0].{name}", element);
+        }
+        public static IQueryable<HtmlNode> GetHtmlNodes(this ISearchContext element, TimeSpan timeout = default)
+        {
+            if (timeout == default)
+                timeout = TimeSpan.FromSeconds(20);
+            var doc = new HtmlDocument();
+            var start = DateTime.Now;
+
+            IQueryable<HtmlNode> nodes = default;
+            string html = "";
+            while (string.IsNullOrEmpty(html) && (DateTime.Now - start) < timeout)
+            {
+                Task.Delay(100).GetAwaiter().GetResult();
+                if (element as IWebElement != null)
+                {
+                    var outerHtml = ((IWebElement)element).GetAttribute("outerHtml");
+                    if (outerHtml == null)
+                        outerHtml = ((IWebElement)element).GetAttribute("outerHTML");
+                    if (outerHtml == null) continue;
+                    html = outerHtml;
+                }
+                if (element as IWebDriver != null)
+                    html = ((IWebDriver)element).PageSource;
+
+                if (html == null) continue;
+                doc.LoadHtml(html);
+                nodes = doc.DocumentNode.Descendants().AsQueryable();
+            }
+            return nodes;
+        }
+        public static HtmlNode GetHtmlNode(this ISearchContext element, Func<HtmlNode, bool> query, int index = 0, TimeSpan timeout = default, ILogger logger = default)
+        {
+            if (timeout == default)
+                timeout = TimeSpan.FromSeconds(20);
+            if (index < 0)
+                throw new ArgumentOutOfRangeException("index must be positive or zero");
```

Comment:
Please double-check all of the exceptions you raise by making sure the message starts with a capital. e.g. `"Index must be positive or zero"`.